app/actors/hyrax/actors/apply_permission_template_actor.rb

```diff
@@ -13,8 +13,8 @@ def create(env)
       private
 
         def add_edit_users(env)
-          return if env.attributes[:admin_set_id].blank?
-          template = Hyrax::PermissionTemplate.find_by!(admin_set_id: env.attributes[:admin_set_id])
+          return if env.attributes[:admin_set_id].to_s.blank?
```

Comment:
AdminSet got set to be `  attribute :id, Valkyrie::Types::ID.optional`

Reply:
@jcoyne `env.attributes[:admin_set_id]` returns a Valkyrie::Types::ID, so we to cast to a string because that's what `Hyrax::PermissionTemplate.find_by!` expects. Would a todo for refactoring `Hyrax::PermissionTemplate.find_by!` suffice?

Reply:
Without the `to_s` you get:
```
ActiveRecord::StatementInvalid:
  can't cast Valkyrie::ID
```

Reply:
but where is that coming from?  Isn't `Environment#attributes` set here: https://github.com/samvera/hyrax/blob/master/app/controllers/concerns/hyrax/works_controller_behavior.rb#L162

Which comes from https://github.com/samvera/hyrax/blob/master/app/controllers/concerns/hyrax/works_controller_behavior.rb#L228 which is basically `ActionController::Parameters`?

Reply:
Looking a the code, the DefaultAdminSetActor will use whatever string the user provided but if that is blank, load in an admin set and that loads a Valkyrie::ID into `:admin_set_id`.

The DefaultAdminSetActor looks ripe to refactor into the work_change_set, I opened #1994, I think we can shut this PR down and do that instead